migrations/2024-12-05-233704_add_person_content_combined_table/up.sql

```diff
@@ -0,0 +1,71 @@
+-- Creates combined tables for
+-- person_content: (comment, post)
+-- person_saved: (comment, post)
+CREATE TABLE person_content_combined (
+    id serial PRIMARY KEY,
+    published timestamptz NOT NULL,
+    post_id int UNIQUE REFERENCES post ON UPDATE CASCADE ON DELETE CASCADE,
+    comment_id int UNIQUE REFERENCES COMMENT ON UPDATE CASCADE ON DELETE CASCADE,
+    -- Make sure only one of the columns is not null
+    CHECK (num_nonnulls (post_id, comment_id) = 1)
+);
+
+CREATE INDEX idx_person_content_combined_published ON person_content_combined (published DESC, id DESC);
+
+CREATE INDEX idx_person_content_combined_published_asc ON person_content_combined (reverse_timestamp_sort (published) DESC, id DESC);
+
+-- Updating the history
+INSERT INTO person_content_combined (published, post_id, comment_id)
+SELECT
+    published,
+    id,
+    NULL::int
+FROM
+    post
+UNION ALL
+SELECT
+    published,
+    NULL::int,
+    id
+FROM
+    comment;
+
+-- This one is special, because you use the saved date, not the ordinary published
+CREATE TABLE person_saved_combined (
+    id serial PRIMARY KEY,
+    published timestamptz NOT NULL,
```

Comment:
Then rename this column to make it clear.

Reply:
Not sure about this one. All of the time columns for all of these combined tables are named published, and that note is probably good enough to denote that its the saved date. Doesn't matter too much to me either way tho.

Reply:
Thats because the other tables actually store the publish date, which is not the case here. And we dont look at migration files after they are merged, so this comment wouldnt be visible from the code at all.

Reply:
I agree, rename this "published" column to "saved"